src/System.Management.Automation/engine/parser/ConstantValues.cs

```diff
@@ -412,6 +448,12 @@ public object VisitPipeline(PipelineAst pipelineAst)
             return pipelineAst.GetPureExpression().Accept(this);
         }
 
+        public object VisitTernaryExpression(TernaryExpressionAst ternaryExpressionAst)
+        {
+            CheckIsConstant(ternaryExpressionAst, "Caller to verify ast is constant");
+            return CompileAndInvoke(ternaryExpressionAst);
+        }
```

Comment:
`CompileAndInvoke` exists mostly because operators are incredibly complicated to constant fold, but this operator is easy to implement without falling back on dynamic code gen.

Reply:
Good point. I updated it to be:
```
object condition = ternaryExpressionAst.Condition.Accept(this);
return LanguagePrimitives.IsTrue(condition)
    ? ternaryExpressionAst.IfTrue.Accept(this)
    : ternaryExpressionAst.IfFalse.Accept(this);
```
Let me know if you think it's not right.